src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java

```diff
@@ -78,7 +78,7 @@
             Pattern.compile("\\s*Violation Message Keys:\\s*");
 
     /** Regular expression for detecting ANTLR tokens(for e.g. CLASS_DEF). */
-    private static final Pattern TOKEN_TEXT_PATTERN = Pattern.compile("([A-Z]+_*+)+[A-Z]+");
+    private static final Pattern TOKEN_TEXT_PATTERN = Pattern.compile("([A-Z]+_?)+[A-Z]+");
```

Comment:
We have patterns with no underscores at all? Can you give an example?

Reply:
`TYPECAST`

Reply:
https://checkstyle.sourceforge.io/config_annotation.html#AnnotationOnSameLine You can find one such token here.

Reply:
`TYPE` `PARAMETERS` `EXPR`

Reply:
`RCURLY` `LCURLY`

Reply:
ah ok.

Then shouldn't the `?` be transferred to the `+` outside the `()`s since the `[A-Z]+` on the end covers it?
Example: `([A-Z]+_)*[A-Z]+`

Reply:
Good point, done.